app/models/user.rb

```diff
@@ -4,7 +4,7 @@ class User < ApplicationRecord
   include PgSearch::Model
   multisearchable against: %i[first_name last_name username note]
   has_many :replies, dependent: :destroy
-  has_many :requests, through: :replies
+  has_many :requests, -> { reorder(created_at: :desc).distinct }, through: :replies
```

Comment:
@roschaefer: The extra `reorder` is necessary, as otherwise, ActiveRecord creates the following SQL query:

```
"SELECT DISTINCT "requests".* FROM "requests" INNER JOIN "replies" ON "requests"."id" = "replies"."request_id" WHERE "replies"."user_id" = 1 ORDER BY "requests"."created_at" DESC, "replies"."created_at" DESC"
```

That’ll cause an SQL error (as `replies.created_at` is not part of the selected columns).

What we actually want is:

```
SELECT DISTINCT "requests".* FROM "requests" INNER JOIN "replies" ON "requests"."id" = "replies"."request_id" WHERE "replies"."user_id" = $1 ORDER BY "requests"."created_at" DESC
```

Don’t think that’s the best solution – and I think not having a default_scope but rather applying even simple ordering explicitly might help…? Thoughts?

Reply:
efff, yes, also why the `distinct`? Is this necessary?

Reply:
If there are two replies to the same request by the same user, without `distinct`, `User.requests` would contain the request two times.